cli/azd/pkg/project/service_target.go

```diff
@@ -83,6 +83,15 @@ func resourceTypeMismatchError(
 	)
 }
 
+// SupportsDelayedProvisioning returns true if the service target kind
+// supports delayed provisioning resources at deployment time, otherwise false.
+//
+// As an example, ContainerAppTarget is able to provision the container app as part of deployment,
+// and thus returns true.
+func (st ServiceTargetKind) SupportsDelayedProvisioning() bool {
```

Comment:
This is to mainly deal with the fact that our project loading logic right now tries to resolve the target for every service and if the resource is not found, things go sideways, right?  Do you imagine long term we'd use this elsewhere?

I think this is a great tactical thing - but I worry that it's a wart that shows us our model is somewhat incomplete in how we think about these types of services where a `deploy` may involve something that looks like a `provision` (maybe more common in k8s land, depending on how we model the concepts when we w get there?) - so I'm trying to better understand where else in the system we might have places where `SupportsDelayedProvisioning` would be used in the future.

Reply:
Chatted with Matt offline.

Essentially, this is a unique problem where:

1. The code deployment involves going through IaC provider
2. The Azure resource itself is tied to the application -- so it shouldn't be "pre-created" prior to the app deployment